include/class.ticket.php

```diff
@@ -3615,5 +3615,33 @@ static function agentActions($agent, $options=array()) {
 
         require STAFFINC_DIR.'templates/tickets-actions.tmpl.php';
     }
+    
+    // Close tickets based on Ticket Auto-Close from config
+    function AutoClose() {
+        global $cfg;
+        
+        $grace = $cfg->getAutoCloseDuration();
+        $graceStatusId = $cfg->getAutoCloseStatusId();
+        
+        if($graceStatusId != 0){ 
+            if($grace == 1){
+                $plural = 'hour';
+                } else {
+                $plural = 'hours';
+            }
+            
+            // select all tickets marked as selected status from config page where updated is older than ($grace) hours ago
+            $sql  = 'SELECT ticket_id FROM '.TICKET_TABLE.' ticket '
```

Comment:
@protich could you help convert this to ORM?

Reply:
Sorry for a late response -  it would be something like below - assuming grace period is in hours.

```php
      $tickets = static::objects()
            ->filter(array(
                'status_id' => $graceStatusId,
                'updated__lt' => SqlFunction::NOW()->minus(SqlInterval::HOUR($grace))
             ))
             ->limit(100);
```

Totally unrested - ;)

Reply:
@protich Thanks for the input. I will test. One other question setstatus() does not accept closed as input, it only accepts the I'd number for the status. Is this right?